docs/book/src/developer/e2e.md

```diff
@@ -0,0 +1,228 @@
+
+Following guidelines should be followed when developing E2E tests:
+
+- Use the [Cluster API test framework].
+- Define test spec reflecting real user workflow, e.g. [Cluster API quick start].
+- Unless you are testing provider specific features, ensure your test can run with
+  different infrastructure providers (see [Writing Portable Tests](#writing-portable-e2e-tests)).
+  
+The [Cluster API test framework] provides you a set of helpers method for getting your test in place
+quickly; the [test E2E package] provide examples of how this can be achieved and reusable 
+test specs for the most common Cluster API use cases.
+
+## Prerequisites
+
+Each E2E test requires a set of artifacts to be available:
+
+- Binaries & docker images for Kubernetes, CNI, CRI & CSI 
+- Manifests & docker images for the Cluster API core components
+- Manifests & docker images for the Cluster API infrastructure provider; in most cases
+  also machine images are required (AMI, OVA etc.)
+- Credentials for the target infrastructure provider 
+- Other support tools (e.g. kustomize, gsutil etc.)
+
+The Cluster API test framework provides support for building and retrieving the manifest
+files for Cluster API core components and for the Cluster API infrastructure provider
+(see [Setup](#setup))
+
+For the remaining tasks you can find examples of
+how this can be implemented e.g. in [CAPA E2E tests] and [CAPG E2E tests]. 
+
+## Setup
+
+In order to run E2E tests it is required to create a Kubernetes cluster with a 
+complete set of Cluster API providers installed. Setting up those elements is 
+usually implemented in a `BeforeSuite` function, and it consists of two steps:
+
+- Defining an E2E config file
+- Creating the management cluster and installing providers
+
+### Defining an E2E config file
+
+The [E2E config file] provides a convenient and flexible way to define common tasks for
+setting up a management cluster.
+
+Using the config file it is possible to:
+
+- Define the list of providers to be installed in the management cluster. Most notably, 
+  for each provider it is possible to define:
+  - One or more versions of the providers manifest (built from the sources, or pulled from a
+    remote location).
+  - A list of additional files to be added to the provider repository, to be used e.g.
+    to provide `cluster-templates.yaml` files.
+- Define the list of variables to be used when doing `clusterctl init` or
+  `clusterctl config cluster`.
+- Define a list of intervals to be used in the test specs for defining timeouts for the
+  wait commands/for the `Eventually` method.
+- Define the list of images to be loaded in the management cluster (this is specif of
+  management cluster based on kind).
+
+An [example E2E config file] can be found here.
+
+<aside class="note">
+
+<h1>Deprecated E2E config file format</h1>
+
+The [Cluster API test framework] includes also a [deprecated E2E config file] implementation,
```

Comment:
We need to add `// Deprecated` command for this struct?

Reply:
yes please, it seems I forgot to do this when doing the framework re-shuffle